hooks/useMaskParam.ts

```diff
@@ -0,0 +1,19 @@
+import { useSearchParams } from 'next/navigation';
+import { useMemo } from 'react';
+
+export function useMaskParam(): boolean {
+    const searchParams = useSearchParams();
+    const mask = searchParams.get('mask');
+
+    return useMemo(() => {
+        if (!mask) {
+            return false;
+        }
+
+        try {
+            return Boolean(JSON.parse(mask));
+        } catch (_) {
+            return false;
+        }
+    }, [mask]);
```

Comment:
Is it worth memo-ing? Booleans are simple scalars, and the computation doesn't seem to be expensive.